web/misc.go

```diff
@@ -168,12 +168,19 @@ func parseIntoBSO(jsonData json.RawMessage, bso *syncstorage.PutBSOInput) *parse
 
 	if r, ok := bkeys["ttl"]; ok {
 		var ttl int
-		err := json.Unmarshal(r, &ttl)
-		if err != nil {
-			return &parseError{bId: bId, field: "ttl", msg: "Invalid format"}
+
+		// bug 1332552, treat `ttl:null` as never expiring
+		// by using 100 years for it (in seconds)
+		if len(r) == 4 && string(r[:]) == "null" {
```

Comment:
I am flabberghasted that not only does Go put you in this situation, but it also forces you to do bytewise array comparison in this case.

Please make sure you have tests for invalid JSON, such as:

```
"ttl": nu,
"ttl": "null",
"ttl": [],
```